app/frontend/src/application/jobseekers/manageQualifications.test.js

```diff
@@ -67,78 +67,70 @@ describe('manageQualifications', () => {
-    });
-  });
-
-  describe('addDeletionEventListener', () => {
-    manageQualifications.onDelete = jest.fn();
-    const onDeleteSpy = jest.spyOn(manageQualifications, 'onDelete');
-    const newRow = document.getElementsByClassName(ROW_CLASS)[0].cloneNode(true);
-    const deleteButton = newRow.lastElementChild;
-    addDeletionEventListener(deleteButton);
-    deleteButton.dispatchEvent(new Event('click'));
-
-    test('when the link is clicked, it deletes row and renumbers remaining rows', () => {
-      expect(onDeleteSpy).toHaveBeenCalledWith(deleteButton);
-    });
-  });
-
   describe('onDelete', () => {
-    manageQualifications.renumberRemainingRows = jest.fn();
+    manageQualifications.renumberRows = jest.fn();
+    const renumberRowsSpy = jest.spyOn(manageQualifications, 'renumberRows');
+
     const initialNumberOfRows = document.getElementsByClassName(ROW_CLASS).length;
     const rowNumberToDelete = '2';
 
     beforeEach(() => {
-      onDelete(document.getElementById(`delete_${rowNumberToDelete}`));
+      onDelete(document.getElementsByClassName(ROW_CLASS)[rowNumberToDelete].children[2]);
     });
 
     test('deletes row', () => {
       expect(document.getElementsByClassName(ROW_CLASS).length).toBe(initialNumberOfRows - 1);
     });
 
-    test('renumbers the rows after the deleted row', () => {
-      const renumberRemainingRowsSpy = jest.spyOn(manageQualifications, 'renumberRemainingRows');
-      expect(renumberRemainingRowsSpy).toHaveBeenCalledWith(rowNumberToDelete);
+    test('renumbers the rows', () => {
+      expect(renumberRowsSpy).toHaveBeenCalledTimes(2);
     });
   });
 
-  describe('renumberRemainingRows', () => {
+  describe('renumber rows', () => {
     manageQualifications.renumberRow = jest.fn();
     const renumberRowSpy = jest.spyOn(manageQualifications, 'renumberRow');
 
-    test('renumbers the rows after the deleted row, keeping the values and errors', () => {
-      const rows = document.getElementsByClassName(ROW_CLASS);
-      renumberRemainingRows(2);
-      expect(renumberRowSpy).toHaveBeenCalledWith(rows[2], 3, true);
-      expect(renumberRowSpy).toHaveBeenCalledWith(rows[3], 4, true);
+    beforeEach(() => {
+      renumberRows();
+    });
+
+    test('renumbers all rows', () => {
+      expect(renumberRowSpy).toHaveBeenCalledTimes(7);
+    });
+  });
+
+  describe('renumber row', () => {
+    manageQualifications.renumberCell = jest.fn();
+    const renumberCellSpy = jest.spyOn(manageQualifications, 'renumberCell');
+
+    const row = document.getElementsByClassName(FIELDSET_CLASSNAME)[0];
+
+    beforeEach(() => {
+      renumberRow(row, 3);
+    });
+
+    test('renumbers all cells', () => {
+      expect(renumberCellSpy).toHaveBeenCalledTimes(11);
+    });
+  });
+
+  describe('renumbers cells', () => {
+    const row = document.getElementsByClassName(FIELDSET_CLASSNAME)[0];
+
+    test('renumbers all cell attributes and labels', () => {
```

Comment:
Can you test that any digits in the value (e.g. Economics 101) are not renumbered?